File: test/fixtures/java-maven/src/main/java/com/example/infrastructure/repository/OrderRepository.java
package com.example.infrastructure.repository;

import com.example.domain.order.Order;
import com.example.domain.order.OrderItem;
import java.util.HashMap;
import java.util.Map;

public class OrderRepository {
    private Map<String, Order> orders = new HashMap<>();

    public void save(Order order) {
        orders.put(order.toString(), order);
    }

    public Order findById(String id) {
        return orders.get(id);
    }
}

File: test/fixtures/swift-project/Sources/Infrastructure/OrderRepository.swift
import Foundation
import Domain

/// In-memory order repository (infrastructure layer)
public class OrderRepository {
    private var storage: [String: Order] = [:]

    public func save(_ order: Order) {
        storage[order.id] = order
    }

    public func findById(_ id: String) -> Order? {
        return storage[id]
    }

    public func findAll() -> [Order] {
        return Array(storage.values)
    }
}
